src/rules/unbalanced-delimiters.ts

```diff
@@ -0,0 +1,132 @@
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import {Rule} from '../rule';
+import {Document} from 'polymer-analyzer/lib/model/model';
+import {ParsedHtmlDocument} from 'polymer-analyzer/lib/html/html-document';
+import {Severity, Warning} from 'polymer-analyzer/lib/warning/warning';
+import {PolymerElement} from 'polymer-analyzer/lib/polymer/polymer-element';
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import * as matchers from '../matchers';
+import * as utils from '../utils';
+
+export class UnbalancedDelimiters implements Rule {
+  public async check(document: Document): Promise<Warning[]> {
+    let warnings: Warning[] = [];
+
+    const parsedHtml = document.parsedDocument;
+
+    // This linter is only relevant to html documents, so exit out unless the
+    // provided document is one.
+    if (!(parsedHtml instanceof ParsedHtmlDocument)) {
+      return warnings;
+    }
+
+    // Address binding expressions only for polymer elements.
+    for (const element of document.getByKind('polymer-element')) {
+      console.assert(element instanceof PolymerElement);
+
+      for (const warning of this._warningsForPolymerElement(
+               parsedHtml, element)) {
+        warnings.push(warning);
+      }
+    }
+
+    return warnings;
+  }
+
+  private _warningsForPolymerElement(
+      parsedHtml: ParsedHtmlDocument,
+      element: PolymerElement): Warning[] {
+    let warnings: Warning[] = [];
+
+    if (!element.domModule) {
+      return warnings;
+    }
+
+    const template = dom5.query(element.domModule, matchers.isTemplate);
+    if (!template) {
+      return warnings;
+    }
+
+    const templateContent =
+        parse5.treeAdapters.default.getTemplateContent(template);
+
+    dom5.nodeWalkAll(templateContent, (node: parse5.ASTNode) => {
+
+      if (dom5.isElement(node) && node.attrs.length > 0) {
+        for (const warning of this._warningsForElementAttrs(parsedHtml, node)) {
+          warnings.push(warning);
+        }
+      }
+      if (dom5.isTextNode(node) && typeof node.value === 'string' &&
+          this._isBadBindingExpression(node.value)) {
+        warnings.push({
+          code: 'unbalanced-delimiters',
+          message: `Expression ${node.value} has unbalanced delimiters`,
+          severity: Severity.ERROR,
+          sourceRange: utils.getSourceRangeForTextNode(parsedHtml.url, node)!
+        });
+      }
+
+      return false;
+    });
+    return warnings;
+  }
+
+  private _warningsForElementAttrs(
+      parsedHtml: ParsedHtmlDocument,
+      element: parse5.ASTNode): Warning[] {
+    const warnings: Warning[] = [];
+    for (const attr of element.attrs) {
+      if (this._extractBadBindingExpression(attr.value || '')) {
+        warnings.push({
+          code: 'unbalanced-delimiters',
+          message: `Expression ${attr
+                       .value
+                   } has unbalanced delimiters in attribute ${attr.name}`,
```

Comment:
Generally I'd leave out information in the message that's expressed well in the sourceRange